### DISCUSSION TITLE
tips on making 2-way hyperlinks between nodes ?

### DISCUSSION THREAD

**Discussion by kenfhill84083** [freeplane/freeplane#1255] [Help] [answered]
In a map, I would like to create reciprocal, 2-way hyperlinks between, for example, node A and node Z.  

In one step, make A link to Z and Z link to A.  Kind of similar to the way connector-linked nodes are mutually navigable to each other (by right-click > goto...).  But without the connector and without having to right-click.  Just normal hyperlinks.

This is not a feature request, it is an information request.

Thanks.

**Comment by EdoFro** [comment]
Hi,
When I need to do that (not very frequently), I use the node pop menu 

![image](https://github.com/freeplane/freeplane/assets/35700575/c7d24cee-d6db-4902-8cb9-3404ff8e9475)

1. Select one node
2. open pop menu (right click on node)
   i. Insert->Link->Set link anchor (Ctrl+Alt+Home)
4. select the other node
5. open pop menu (right click on node)
   i. Insert->Link->Make link to anchor (Ctrl+Alt+End)
   ii. Insert->Link->Make link from anchor

I don't remember if the hotkeys there are standard or if I set them.

I don't know if there is an easier way (without scripting)

Hope it helps,
edo

**Reply by kenfhill84083** [reply]
Thanks.  I think I found I little quicker way with command search and/or hotkey.
1. select A then select Z
2. follow Insert->Link->Add local hyperlink (Alt+Shift+L [here])
3. select Z then select A
4. follow Insert->Link->Add local hyperlink (Alt+Shift+L [here])

follow Insert->Link->Add local hyperlink seems to make the 1st-selected node the link and the 2nd-selected node the target.  So you have to reverse the selection order.  

Not a single step but I had envisioned having to copy-paste node IDs so this is streamlined vs my supposition.

Thanks for pointing me to the right-click menu - I should tattoo "right-click" on one wrist and "command search" on the other.

**Reply by EdoFro** [reply]
What I like about using Link Anchor is that you can set it in one node and close that branch and navigate to other end of the map and then do the linking. You don't have to worry if the first node is still selected or not.

You can set the anchor in one view and link a node from another view (for the same map or another one).

**Reply by macmarrum** [reply, marked as answer]
A good point, Edo :+1: 

1. Set anchor
2. Select the target node
3. [insertTwoWayLinkUsingAnchor.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/insertTwoWayLinkUsingAnchor.groovy)

**Comment by macmarrum** [comment]
This script should do the trick in one go: [insertTwoWayLink.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/insertTwoWayLink.groovy)

**Reply by kenfhill84083** [reply]
@macmarrum Do you just throw these amazing, useful scripts together instantaneously? 

Thanks.  That is exactly what I was looking for and works perfectly.

**Reply by EdoFro** [reply]
Just an idea so this script replaces the ctrl+alt+L command:
- if executed one time, it creates just the "to" link
- if executed a second time it adds the "from" or "back" link

```groovy
// @ExecutionModes({ON_SINGLE_NODE="/menu_bar/Mac1/Insert"})
// https://github.com/freeplane/freeplane/discussions/1255

def nodes = c.selecteds.collect()
if (nodes.size() == 2) {
    if( nodes[0].link.node== nodes[1]){
            nodes[1].link.node = nodes[0]
        } else {
            nodes[0].link.node = nodes[1]
        }
} else {
    c.statusInfo = "got ${nodes.size()} nodes - expected 2"
}
```

if you reasign the ctrl+shift+L shortcut to it, it has a very natural usage.

**Comment by kenfhill84083** [comment]
> What I like about using Link Anchor is that you can set it in one node and close that branch and navigate to other end of the map and then do the linking. You don't have to worry if the first node is still selected or not.
> 
> You can set the anchor in one view and link a node from another view (for the same map or another one).

Good points.  I was confused by the terminology (still am) but maybe playing around with it might cement it in my old brain - unless I only wind up doing it occasionally so I can forget again and have to figure it out again.  

Between the 2 suggestions made here, I should be set.  Thanks.
